cmd/cli/app/provider/provider_enroll.go

```diff
@@ -83,24 +83,25 @@ func EnrollProviderCommand(ctx context.Context, cmd *cobra.Command, conn *grpc.C
 		}
 	}
 
-	prov, err := provcli.GetProvider(ctx, &minderv1.GetProviderRequest{
-		Context: &minderv1.Context{Provider: &provider, Project: &project},
-		Name:    provider,
+	prov, err := provcli.GetProviderClassDefinition(ctx, &minderv1.GetProviderClassDefinitionRequest{
+		Context: &minderv1.Context{
+			Project: &project,
+		},
+		Class: provider,
 	})
 	if err != nil {
 		return cli.MessageAndError("Error getting provider", err)
 	}
 
 	if token != "" {
-		if !prov.Provider.SupportsAuthFlow(minderv1.AuthorizationFlow_AUTHORIZATION_FLOW_USER_INPUT) {
+		if !prov.Definition.SupportsAuthFlow(minderv1.AuthorizationFlow_AUTHORIZATION_FLOW_USER_INPUT) {
```

Comment:
if we're creating the provider in one go with the right auth flow, why check it here even? Let's just try to use the provider and fail from the server side. It should just display to the user as an error message. that would also simplify the logic on the CLI.